x/trigger/simulation/operations_test.go

```diff
@@ -0,0 +1,147 @@
+package simulation_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/x/bank/testutil"
+
+	"github.com/provenance-io/provenance/app"
+	simappparams "github.com/provenance-io/provenance/app/params"
+	"github.com/provenance-io/provenance/x/trigger/simulation"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+type SimTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *app.App
+}
+
+func (s *SimTestSuite) SetupTest() {
+	s.app = app.Setup(s.T())
+	s.ctx = s.app.BaseApp.NewContext(false, tmproto.Header{})
+}
+
+func (s *SimTestSuite) TestWeightedOperations() {
+	cdc := s.app.AppCodec()
+	appParams := make(simtypes.AppParams)
+
+	weightedOps := simulation.WeightedOperations(appParams, cdc, s.app.TriggerKeeper,
+		s.app.AccountKeeper, s.app.BankKeeper,
+	)
+
+	// setup 3 accounts
+	source := rand.NewSource(1)
+	r := rand.New(source)
+	accs := s.getTestingAccounts(r, 3)
+
+	// begin a new block
+	s.app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: s.app.LastBlockHeight() + 1, AppHash: s.app.LastCommitID().Hash}})
+
+	expected := []struct {
+		weight     int
+		opMsgRoute string
+		opMsgName  string
+	}{
+		{simappparams.DefaultWeightSubmitCreateTrigger, sdk.MsgTypeURL(&types.MsgCreateTriggerRequest{}), sdk.MsgTypeURL(&types.MsgCreateTriggerRequest{})},
+		{simappparams.DefaultWeightSubmitDestroyTrigger, sdk.MsgTypeURL(&types.MsgDestroyTriggerRequest{}), sdk.MsgTypeURL(&types.MsgDestroyTriggerRequest{})},
+	}
+
+	for i, w := range weightedOps {
```

Comment:
Take a look at the changes I made to similar tests in #1561. E.g. https://github.com/provenance-io/provenance/blob/55c1ff996fee7aac0bd693228319f400e1c49015/x/attribute/simulation/operations_test.go#L90-L112

It'd be nice to use that pattern moving forward as it makes the tests easier to understand when it fails.

Similarly, I tweaked the pattern of the `TestSimulateMsg...` tests to be friendlier when things go wrong, e.g. logging the operation message so that we know what it was when there's a test failure, never ignoring returned errors, and better `msgAndArgs` values.